src/zerocoin.cpp

```diff
@@ -1064,6 +1098,17 @@ set<CBlockIndex *> CZerocoinState::RecalculateAccumulators(CChain *chain) {
     return changes;
 }
 
+bool CZerocoinState::AddSpendToMempool(const vector<CBigNum> &coinSerials, uint256 txHash) {
+    BOOST_FOREACH(CBigNum coinSerial, coinSerials){
+        if (IsUsedCoinSerial(coinSerial) || mempoolCoinSerials.count(coinSerial))
+            return false;
+
+        mempoolCoinSerials[coinSerial] = txHash;        
```

Comment:
Remove `mempoolCoinSerials.count(coinSerial)` from the above line and replace this line with the following code should be better:

```cpp
auto r = mempoolCoinSerials.insert(std::make_pair(coinSerial, txHash));
if (!r.second) {
    return false;
}
```

The benefits with this way is:

1. Hash for `coinSerial` need to calculate only one time instead of two.
2. Lookup need to do only one time instead of two.
3. A new value of `uint256` will construct with copy constructor instead of construct with default constructor then invoke assignment operator. Most of the time copy constructor is efficient than assignment operator.